src/tests/class_tests/openms_gui/source/GUI/TVPrefDialog_test.cpp

```diff
@@ -0,0 +1,352 @@
+  p.setValue("add_losses", "false");
+  p.setValue("sort_by_position", "true");
+  p.setValue("add_precursor_peaks", "true");
+  p.setValue("add_all_precursor_charges", "false");
+  p.setValue("add_abundant_immonium_ions", "true");
+  p.setValue("add_first_prefix_ion", "false");
+  for (String s : {'a', 'b', 'c', 'x', 'y', 'x', 'z'})
+  {
+    p.setValue("add_" + s + "_ions", "true");
+    p.setValue(s + "_intensity", 0.8);
+  }
+  p.setValue("relative_loss_intensity", 0.2);
+  p.setValue("precursor_intensity", 1.0);
+  p.setValue("precursor_H2O_intensity", 0.95);
+  p.setValue("precursor_NH3_intensity", 0.9);
+
+  dialog_.setParam(p);
+  UI->tolerance->setValue(0.2); // tolerance: 0.2
+  UI->unit->setCurrentIndex(0); // Dalton
+
+  // accept the dialog to save the parameters
+  QTest::mouseClick(UI->buttonBox->button(QDialogButtonBox::Ok), Qt::LeftButton);
+
+  // get parameters
+  Param dialog_param = dialog_.getParam();
+
+  // check validity
+#define PARAM(a) dialog_param.getValue(a)
+
+  QVERIFY2(PARAM("default_path") == "C:\dev", "'Default path' param value not exported correctly.");
+  QVERIFY2(PARAM("default_path_current") == "false", "'Use current path' param value not exported correctly.");
+  QVERIFY2(PARAM("use_cached_ms1") == "true", "'Cache ms1 spectra' param value not exported correctly.");
+  QVERIFY2(PARAM("use_cached_ms2") == "false", "'Cache ms2 spectra' param value not exported correctly.");
+  QVERIFY2(PARAM("default_map_view") == "3d", "'Default map view' param value not exported correctly.");
+  QVERIFY2(PARAM("intensity_cutoff") == "on", "'Itensity cutoff' param value not exported correctly.");
+  QVERIFY2(PARAM("on_file_change") == "update automatically", "'Action when file changes' param value not exported correctly.");
+
+  QVERIFY2(PARAM("1d:peak_color") == "#ff0000", "'1D peak color' param value not exported correctly.");
+  QVERIFY2(PARAM("1d:highlighted_peak_color") == "#8b0000", "'1D highlighted peak color' param value not exported correctly.");
+  QVERIFY2(PARAM("1d:icon_color") == "#660000", "'1D icon color' param value not exported correctly.");
+
+  QVERIFY2(PARAM("2d:dot:gradient") == "Linear|0,#ffaa00;6,#ff0000;14,#aa00ff;23,#5500ff;100,#000000", "'2D peak gradient' param value not exported correctly.");
+  QVERIFY2(PARAM("2d:mapping_of_mz_to") == "x_axis", "'2D mapping of mz to' param value not exported correctly.");
+  QVERIFY2(PARAM("2d:dot:feature_icon") == "circle", "'2D feature icon' param value not exported correctly.");
+  QVERIFY2(int(PARAM("2d:dot:feature_icon_size")) == 5, "'2D feature icon size' param value not exported correctly.");
+
+  QVERIFY2(PARAM("3d:dot:gradient") == "Linear|0,#ffea00;6,#ffaa00;100,#ff0000", "'3D peak gradient' param value not exported correctly.");
+  QVERIFY2(int(PARAM("3d:dot:shade_mode")) == 1, "'3D shade mode' param value not exported correctly.");
+  QVERIFY2(int(PARAM("3d:dot:line_width")) == 2, "'3D line width' param value not exported correctly.");
+
```

Comment:
(you are going to hate this): technically you need to test two completely orthogonal sets of values, just to make sure the Dialog does not always return the same value for some widget (despite setting another value in the GUI), and you happen to test exactly for that value.

Reply:
hmmm I see...
Well, at least that's easy to do.